### PR TITLE
Adding dockerfile support #352 and cleanup

### LINKED ISSUE #352: Support finding the `Dockerfile` or `dockerfile` in repos as well
For starters, lets only use the ones from the root of the repo, but they also might work inside of folders, not sure (you can place an action.yml in a folder as well).

Additional code needs to expand [this search](https://github.com/devops-actions/load-available-actions/blob/1f6517b56f7343b399bc09ddcc9605d2dfebc92e/src/main.ts#L218-L219) and then switch on the file. Pick up the designated labels for the Action as information.

### PR DESCRIPTION
**Pull request by GetTuh**
Closes #352

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -1,32 +1,105 @@
...
+        try {
+          // Read the contents of the Dockerfile
+          const data = await promisify(fs.readFile)(item, 'utf8')
+          const label = 'LABEL com.github.actions.'
+          // Check if the Dockerfile has actionable properties
+          if (
+            data.includes(`${label}name=`) &&
+            data.includes(`${label}description=`)
+          ) {
+            core.info(`${item} has dockerfile as an action!`)
```
Comment (rajbos) [inline src/utils.ts:82]: Add brackets to match the rest of the code
```suggestion
            core.info(`[${item}] has dockerfile as an action!`)
```

File: src/utils.ts [hunk 2]
```diff
@@ -1,32 +1,105 @@
...
+  [key: string]: string | undefined
+  name?: string
+  description?: string
+  // icon?: string
+  // color?: string
+
+  // Icon and color is needed for using dockerfiles as actions, but it's not used in the marketplace.
+}
+
+export const returnActionableDockerFiles = async (path: string) => {
```
Comment (rajbos) [inline src/utils.ts:59]: Rename method to match the other methods in this class (don't forget to update the call to this method as well
```suggestion
export const getActionableDockerFiles = async (path: string) => {
```

**Review by rajbos — COMMENTED** [review]
Looking good

File: src/main.ts [hunk 3]
```diff
@@ -348,7 +351,7 @@ function cloneRepo(repo: string, owner: string): string {
     // create a temp directory
     const repoPath = 'actions'
     if (fs.existsSync(repoPath)) {
-      core.info('folder already exists, deleting')
+      // core.info('folder already exists, deleting')
```
Comment (rajbos) [inline src/main.ts:368]: Shall we convert that to a debug message then?

Comment (GetTuh) [reply src/main.ts]: Done

File: src/utils.ts (hunk 1 above)
Comment (GetTuh) [reply src/utils.ts]: Done

**Comment by GetTuh** [comment]
Successful run:
https://github.com/GetTuh/load-available-actions/actions/runs/4809564438
The actions-user.json contains all the necessary data for pnpm which only has a dockerfile, no yaml.

Additionally few code optimizations has been done. It could be further optimized for no technical debt, but it's beyond the scope of my ticket.

**Review by rajbos — APPROVED** [review]
I'll make some touch ups after the merge, but thanks for adding this!

**Comment by rajbos** [comment]
Cool! I've forked that repo to my own org as well, so we can continue testing things there as well.

File: src/utils.ts (hunk 2 above)
Comment (rajbos) [reply src/utils.ts:62]: Updated it in a later commit

**Pull request merged**